### PR TITLE
Include calendars in all bundles

### LINKED ISSUE #4883: no calendar on partial bundles
I am trying to get the connectgaps property to work in a React app where I use a custom plotly.js bundle to reduce the build size. 

With this custom bundle which imports only Surface, connectgaps does not work.

If I use the full plotly.js bundle, the plot works properly. 

See attached github repo for a simple reproduction showing the two plots side-by-side. 
image
[Github repo for reproduction](https://github.com/mattcorner/plotly-surface-connectgaps-error)

I presume that I either need to import something extra in the custom bundle, or that lib/Surface is missing a dependency?

### PR DESCRIPTION
**Pull request by archmoj**
Fixes #4883.

@plotly/plotly_js

### CODE REVIEW

File: lib/register_extra.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        require('./groupby'),
+        require('./sort')
+    ]);
+
+    // components
+    Plotly.register([
+        require('./calendars')
+    ]);
+
+    return Plotly;
```
Comment (alexcjohnson) [inline lib/register_extra.js:34]: This `return Plotly` surprised me. Why not just:

```js
require('./register_extra')(Plotly);
module.exports = Plotly;
```

Comment (archmoj) [reply lib/register_extra.js]: Revised in https://github.com/plotly/plotly.js/pull/5422/commits/1fc1c035b16672e4d314ac1b55fcbc3fce308714.

**Review by alexcjohnson — APPROVED** [review]
Very nice simplification, to put all the common extras in one new file. Nice and 🌴 without stopping anyone from making their own bundle picking just a few of the extras. 💃

**Comment by archmoj** [comment]
Interestingly I get an error right after https://github.com/plotly/plotly.js/pull/5422/commits/1fc1c035b16672e4d314ac1b55fcbc3fce308714 
cc: #5382 now in `mapbox` bundle.

Screenshot from 2021-01-19 19-15-50

@alexcjohnson 
Do you mind if I revert that commit back to the functional form?

I think we need to deal with this kind of problems, until we could start using `d3-geo` cc: #5112.

**Comment by archmoj** [comment]
> Interestingly I get an error right after [1fc1c03](https://github.com/plotly/plotly.js/commit/1fc1c035b16672e4d314ac1b55fcbc3fce308714)
> cc: #5382 now in `mapbox` bundle.
> 
> Screenshot from 2021-01-19 19-15-50
> 
> @alexcjohnson
> Do you mind if I revert that commit back to the functional form?
> 
> I think we need to deal with this kind of problems, until we could start using `d3-geo` cc: #5112.

I also suspect that the issue is caused by https://github.com/plotly/plotly.js/blob/master/tasks/compress_attributes.js

**Comment by archmoj** [comment]
> I also suspect that the issue is caused by https://github.com/plotly/plotly.js/blob/master/tasks/compress_attributes.js

In that case #5425 may help fix the problem for now.

**Comment by alexcjohnson** [comment]
> I also suspect that the issue is caused by https://github.com/plotly/plotly.js/blob/master/tasks/compress_attributes.js

Great idea - that's actually plausible! Regardless, it's clear any change could make it reappear until we fix the root cause. So let's see what happens after #5425 is merged, but if and when it pops up again I think we should dig into it rather than working around it.

**Pull request merged**